### DISCUSSION TITLE
Run a DAG the Last Three Days of the Month

### DISCUSSION THREAD

**Discussion by dcloninger** [apache/airflow#15021] [Q&A]
So I found where the L can be used to indicate the last day of the month.  I have taken a couple shots at scheduling the last three days of the month with no success.  I have only been working with airflow a month or so.  Is this something someone knows how to do, or is this too difficult to implement?  I tried putting the following:

```
schedule_interval='0 9 L,L-1,L-2 * *'
```

But since a dash means an interval (instead of subtracting the number of days) it would not work.  Any ideas would be appreciated.

**Comment by turbaszek** [comment]
Please see the AIP, it should serve your purpose once implemented:
https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-39+Richer+scheduler_interval

What you can currently do is to use branch operator and do:
```
skip = DummyOperator(...)
check_if_should_run = YourCustomBranchOperator(...)

check_if_should_run >> [run_branch, skip]
```
Run this daily and it should work as expected.

**Reply by dcloninger** [reply]
kludged it together this way:

passed this in:
```
op_args=['{{ macros.ds_format(macros.ds_add(ds, 3), "%Y-%m-%d", "%d") }}']
```

checked this way:
```
if input_date in ['01','02','03']:
   run_my_command
```

then I can run the DAG the 26-31 so it will cover leap years
not the prettiest, but it should work...

**Reply by turbaszek** [reply]
If you wan't to use it in more than one DAG you can use `dag_policy` and add your new macro function to user defined macros
